astropy/wcs/wcsapi/fitswcs.py

```diff
@@ -420,8 +420,12 @@ def _get_components_and_classes(self):
             else:
 
                 earth_location = EarthLocation(*self.wcs.obsgeo[:3], unit=u.m)
-                obstime = Time(self.wcs.mjdobs, format='mjd', scale='utc',
-                               location=earth_location)
+                if np.isnan(self.wcs.mjdavg):
+                    obstime = Time(self.wcs.mjdobs, format='mjd', scale='utc',
```

Comment:
Does MJD actually use this scale argument? If so shouldn't we be pulling it from ``TIMESYS``?

(I don't really understand MJD as I have never had to use it).

Reply:
I don't understand it either but according to the `Time` documentation it needs a scale. Pulling it from `TIMESYS` is better. I'll add a fall back to `utc` because JWST files have the `TIMESYS` keywords in the primary header, while the rest of the WCS keywords are in the `SCI` extension header.